lms/djangoapps/mobile_api/decorators.py

```diff
@@ -37,6 +40,9 @@ def _wrapper(self, request, *args, **kwargs):
                         depth=depth,
                         check_if_enrolled=True,
                     )
+                except CourseAccessRedirect:
+                    # Raise a 404 if the user does not have course access
+                    raise Http404
                 except CoursewareAccessException as error:
                     return Response(data=error.to_json(), status=status.HTTP_404_NOT_FOUND)
```

Comment:
@andy-armstrong: This isn't part of your PR, but I wonder about `data=error.to_json()`.  I don't think we should leave lingering security issues, if we think this might be one.  Should we get another opinion?  Do you know if this gives anything away in the actual response?